### PR TITLE
Changes to make inlined pinvoke codegen look more like the fragile NI codegen, for better performance.

### PR DESCRIPTION
**Pull request by fadimounir**
These changes are *only* for System.Private.CoreLib, and any other assembly within the same bubble. The assumption here is that assemblies in this bubble will always ship alongside a CoreCLR runtime, so it's ok to have fragile NI codegen for these scenarios since we'll always have compatible runtime bits.

- Added a JIT interface method to check if two modules are in the same bubble (mostly plumbing to reach the algorithm in jitinterface.cpp)
- Minor fixes to R2RDump, and adding support for new fixup types.

### CODE REVIEW

**Comment by fadimounir** [comment]
@jkotas @davidwrighton I'd like to get your initial feedback on this PR. I haven't done any perf testing yet

File: src/inc/corinfo.h [hunk 1]
```diff
@@ -2305,6 +2305,11 @@ class ICorStaticInfo
             CORINFO_MODULE_HANDLE   scope
             ) = 0;
 
+    virtual BOOL isInSameVersionBubble(
```
Comment (jkotas) [inline src/inc/corinfo.h]: The concept of the version bubble should not be explosed on the JIT/EE interface. I do not understand why you have added this - I do not see any call from JIT.

**Comment by jkotas** [comment]
This does not look like a good change to me. It adds yet another path for what PInvokes can be compiled into; and the benefit is pretty limited - especially given the tiered JITing.

**Comment by fadimounir** [comment]
Ok I think I got confused by where `IsInSameVersionBubble` was implemented (I thought it was in the clrjit library, not crossgen), and I added a whole bunch of plumbing for no reason

File: src/inc/corinfo.h (hunk 1 above)
Comment (fadimounir) [reply src/inc/corinfo.h]: Removed. I was trying to test for version bubble from zapinfo.cpp. I didn't realize the implementation of the `IsInSameVersionBubble` was already compiled into crossgen

File: src/jit/lower.cpp [hunk 2]
```diff
@@ -3605,6 +3605,7 @@ void Lowering::InsertPInvokeCallProlog(GenTreeCall* call)
     }
     else
     {
+#ifndef FEATURE_READYTORUN_COMPILER
```
Comment (jkotas) [inline src/jit/lower.cpp]: Why is this needed?

**Comment by fadimounir** [comment]
The changes are cleaner and more contained now.

> and the benefit is pretty limited - especially given the tiered JITing.
I still have to measure the benefit. My understanding is tiered JITing will help for the stable state, but we still have a significant startup time regression. I want to see what the impact of these changes would be. These changes will inline the calls to the two JIT Pinvoke helpers, so it might help at startup...

File: src/jit/lower.cpp (hunk 2 above)
Comment (fadimounir) [reply src/jit/lower.cpp]: The call to `embedMethodHandle` is not R2R compatible (There's an assert to make sure we don't call it). I need to look into this further.
But also, i'm not sure why we need it for pinvoke. It seems to work without it

**Comment by jkotas** [comment]
> These changes will inline the calls to the two JIT Pinvoke helpers, so it might help at startup...

I am pretty sure that these changes are not going to have measurable impact for startup.

**Comment by jkotas** [comment]
Closing stale PRs
